flask/tests/test_datasets.py

```diff
@@ -210,10 +211,35 @@ def test_delete_dataset_admin(client, test_database, login_as):
     assert client.delete("/api/datasets/1").status_code == 204
     assert client.get("/api/datasets/1").status_code == 404
 
+    # Check length of permission groups
+    assert (len(dataset.groups)) == 0
+
     # Deleting a nonexistent dataset
     assert client.delete("/api/datasets/400").status_code == 404
     assert client.get("/api/datasets/400").status_code == 404
 
+def test_delete_dataset_with_tissue_sample(client, test_database, login_as):
+    """
+    DELETE /api/datasets/:id as an administrator
+    """
+    login_as("admin")
+
+    # Deleting a dataset without an analysis but associated with tissue sample
+    dataset_5 = (
+        models.Dataset.query.filter(models.Dataset.dataset_id == 5)
+        .options(
+            joinedload(models.Dataset.tissue_sample).joinedload(
+                models.TissueSample.datasets
+            ),
+        )
```

Comment:
Last thing: you don't need this join here since you're only looking for `tissue_sample_id`, which is already a column in the dataset table.

Reply:
That's right!